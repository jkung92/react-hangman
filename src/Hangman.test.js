import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Hangman from './Hangman';
import img0 from './0.jpg';
import img1 from './1.jpg';

// smoke test
it('renders without crashing', function() {
  shallow(<Hangman />);
});

// snapshot test
it('matches snapshot', function() {
  let wrapper = shallow(<Hangman />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

it('simulates img changes after incorrect guess', function() {
  let wrapper = shallow(<Hangman />);
  wrapper.setState({ answer: 'apple' });
  const incorrectButton = wrapper.find("button[value='b']");
  incorrectButton.simulate('click', { target: { value: 'b' } });
  expect(wrapper.state().nWrong).toBe(1);
  expect(wrapper.state().guessed.has('b')).toBe(true);
  // This is how to target the "img" element
  const img = wrapper.find('img');
  expect(img.props().src).toBe('1.jpg');
});

it('simulates img does not change after correct guess', function() {
  let wrapper = shallow(<Hangman />);
  wrapper.setState({ answer: 'apple' });
  const correctButton = wrapper.find("button[value='a']");
  // console.log(`This is t dhe correct btn`, correctButton.debug());
  correctButton.simulate('click', { target: { value: 'a' } });
  expect(wrapper.state().nWrong).toBe(0);
  expect(wrapper.state().guessed.has('a')).toBe(true);
  // This is how to target the "img" element
  const img = wrapper.find('img');
  expect(img.props().src).toBe('0.jpg');
});

it('simulates ends game after maxGuesses has been exceeded', function() {
  let wrapper = shallow(<Hangman />);
  // Set nWrong to 5
  wrapper.setState({ answer: 'apple' });
  wrapper.setState({ nWrong: 5 });
  const incorrectButton = wrapper.find("button[value='d']");
  incorrectButton.simulate('click', { target: { value: 'd' } });
  // check if nWrong === maxGuesses expect img src = img 6.jpg
  const img = wrapper.find('img');
  expect(img.props().src).toBe('6.jpg');
  const msg = wrapper.find('#win-or-lose');
  expect(msg.html()).toBe('<p id="win-or-lose">You lose! Word was: apple</p>');
});

it('resets game after reset button has been pressed', function() {
  let wrapper = shallow(<Hangman />);
  const incorrectButton = wrapper.find('#reset-btn');
  incorrectButton.simulate('click');
  expect(wrapper.state().nWrong).toBe(0);
  expect(wrapper.state().guessed.size).toBe(0);
});
