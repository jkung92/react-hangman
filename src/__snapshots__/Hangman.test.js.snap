// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="Hangman"
>
  <img
    src="0.jpg"
  />
  <p>
     Number wrong:
    0
  </p>
  <p
    className="Hangman-word"
  >
    _
    _
    _
    _
    _
    _
  </p>
  <p
    id="win-or-lose"
  >
    <button
      disabled={false}
      id="btn"
      key="a"
      onClick={[Function]}
      value="a"
    >
      a
    </button>
    <button
      disabled={false}
      id="btn"
      key="b"
      onClick={[Function]}
      value="b"
    >
      b
    </button>
    <button
      disabled={false}
      id="btn"
      key="c"
      onClick={[Function]}
      value="c"
    >
      c
    </button>
    <button
      disabled={false}
      id="btn"
      key="d"
      onClick={[Function]}
      value="d"
    >
      d
    </button>
    <button
      disabled={false}
      id="btn"
      key="e"
      onClick={[Function]}
      value="e"
    >
      e
    </button>
    <button
      disabled={false}
      id="btn"
      key="f"
      onClick={[Function]}
      value="f"
    >
      f
    </button>
    <button
      disabled={false}
      id="btn"
      key="g"
      onClick={[Function]}
      value="g"
    >
      g
    </button>
    <button
      disabled={false}
      id="btn"
      key="h"
      onClick={[Function]}
      value="h"
    >
      h
    </button>
    <button
      disabled={false}
      id="btn"
      key="i"
      onClick={[Function]}
      value="i"
    >
      i
    </button>
    <button
      disabled={false}
      id="btn"
      key="j"
      onClick={[Function]}
      value="j"
    >
      j
    </button>
    <button
      disabled={false}
      id="btn"
      key="k"
      onClick={[Function]}
      value="k"
    >
      k
    </button>
    <button
      disabled={false}
      id="btn"
      key="l"
      onClick={[Function]}
      value="l"
    >
      l
    </button>
    <button
      disabled={false}
      id="btn"
      key="m"
      onClick={[Function]}
      value="m"
    >
      m
    </button>
    <button
      disabled={false}
      id="btn"
      key="n"
      onClick={[Function]}
      value="n"
    >
      n
    </button>
    <button
      disabled={false}
      id="btn"
      key="o"
      onClick={[Function]}
      value="o"
    >
      o
    </button>
    <button
      disabled={false}
      id="btn"
      key="p"
      onClick={[Function]}
      value="p"
    >
      p
    </button>
    <button
      disabled={false}
      id="btn"
      key="q"
      onClick={[Function]}
      value="q"
    >
      q
    </button>
    <button
      disabled={false}
      id="btn"
      key="r"
      onClick={[Function]}
      value="r"
    >
      r
    </button>
    <button
      disabled={false}
      id="btn"
      key="s"
      onClick={[Function]}
      value="s"
    >
      s
    </button>
    <button
      disabled={false}
      id="btn"
      key="t"
      onClick={[Function]}
      value="t"
    >
      t
    </button>
    <button
      disabled={false}
      id="btn"
      key="u"
      onClick={[Function]}
      value="u"
    >
      u
    </button>
    <button
      disabled={false}
      id="btn"
      key="v"
      onClick={[Function]}
      value="v"
    >
      v
    </button>
    <button
      disabled={false}
      id="btn"
      key="w"
      onClick={[Function]}
      value="w"
    >
      w
    </button>
    <button
      disabled={false}
      id="btn"
      key="x"
      onClick={[Function]}
      value="x"
    >
      x
    </button>
    <button
      disabled={false}
      id="btn"
      key="y"
      onClick={[Function]}
      value="y"
    >
      y
    </button>
    <button
      disabled={false}
      id="btn"
      key="z"
      onClick={[Function]}
      value="z"
    >
      z
    </button>
  </p>
  <button
    id="reset-btn"
    onClick={[Function]}
  >
     
    Reset Game
  </button>
</div>
`;
